backend/spec/requests/roles_request_spec.rb

```diff
@@ -2,14 +2,42 @@
 
 RSpec.describe 'RolesController', type: :request do
   describe '#list' do
-    let!(:role) { create(:role) }
+    let!(:role_moderator) { create(:role_moderator) }
+    let!(:role_admin) { create(:role_admin) }
+    let!(:role_teacher) { create(:role_teacher) }
+    let!(:role_student) { create(:role_student) }
+    let!(:all_roles) do
+      [{ 'id' => role_moderator.id, 'role_type' => role_moderator.role_type }, { 'id' => role_admin.id, 'role_type' => role_admin.role_type },
+       { 'id' => role_teacher.id, 'role_type' => role_teacher.role_type }, { 'id' => role_student.id, 'role_type' => role_student.role_type }]
+    end
+    let!(:not_admin_roles) do
```

Comment:
create not_admin_roles first and make all_roles = not_admin_roles  + role_admin

Reply:
Oh true!